Utilities/sysinfo.cpp

```diff
@@ -33,8 +33,25 @@ bool utils::has_avx2()
 
 bool utils::has_rtm()
 {
-	// Check RTM and MPX extensions in order to filter out TSX on Haswell CPUs
-	static const bool g_value = get_cpuid(0, 0)[0] >= 0x7 && (get_cpuid(7, 0)[1] & 0x4800) == 0x4800;
+	// Check RTM
+	static bool g_value = get_cpuid(0, 0)[0] >= 0x7 && (get_cpuid(7, 0)[1] & 0x800) == 0x800;
+	
+	// Do these checks only if TSX is available and only once
+	static bool RTMChecked = !g_value;
+	if (RTMChecked)
+	{
+		return g_value;
+	}
+	else
+	{
+		// Get CPU Model
+		int Model = (get_cpuid(1, 0)[0] >> 4) & 0xf;
+		Model += ((get_cpuid(1, 0)[0] >> 16) & 0xf) << 4;
+		
+		// If CPU is Haswell, disable TSX (for now)
+		g_value = Model != 0x3c && Model != 0x3f && Model != 0x45 && Model != 0x46;
+		RTMChecked = !RTMChecked;
```

Comment:
Use snake_case for consistency with the rest of the file and compliance with the emulator coding style guidelines. Minor nit really, only `Model` and `RTMChecked` need to be changed